torch/distributions/constraints.py

```diff
@@ -228,6 +230,23 @@ def __repr__(self):
         return fmt_string
 
 
+class _OpenInterval(Constraint):
+    """
+    Constrain to a real interval `(lower_bound, upper_bound)`.
+    """
+    def __init__(self, lower_bound, upper_bound):
+        self.lower_bound = lower_bound
+        self.upper_bound = upper_bound
+
+    def check(self, value):
+        return (self.lower_bound < value) & (value < self.upper_bound)
+
+    def __repr__(self):
+        fmt_string = self.__class__.__name__[1:]
+        fmt_string += '(lower_bound={}, upper_bound={})'.format(self.lower_bound, self.upper_bound)
+        return fmt_string
+
+
```

Comment:
I'd rather avoid introducing the notion of open/closed constraints in this library, since most of the algorithms cannot distinguish open vs closed, and introducing the distinction could lead to a proliferation of incompatible types. Do you have a specific reason for using `open_interval` rather than the existing `interval`?

Reply:
Thanks for the input! I'll write the tests and remove the custom entropy method to rely on the one from ExponentialFamily. As for the open interval, the issue I was trying to avoid is that the distribution is not properly defined when probs=0 or probs=1 (as evidenced by all the log(probs) and log(1-probs) terms appearing throughout the code). However clamping should work just fine, I will also change this.